### PR TITLE
[NCL-8250]: Update DelAn BPM process

### PR DESCRIPTION
**Pull request by akridl**

### CODE REVIEW

**Comment by akridl** [comment]
retest this please

File: facade/src/main/java/org/jboss/pnc/facade/deliverables/api/AnalysisResult.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import lombok.Builder;
 import lombok.Value;
 import org.jboss.pnc.api.deliverablesanalyzer.dto.FinderResult;
+import org.jboss.pnc.api.dto.Request;
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/deliverables/api/AnalysisResult.java:23]: standalone import?

File: rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/DeliverableAnalysisEndpointImpl.java [hunk 2]
```diff
@@ -38,9 +47,27 @@ public class DeliverableAnalysisEndpointImpl implements DeliverableAnalysisEndpo
     @Inject
     private DeliverableAnalyzerOperationMapper deliverableAnalyzerOperationMapper;
 
+    @Inject
+    private ManagedExecutorService executorService;
+
     @Override
     public void completeAnalysis(AnalysisResult response) {
-        resultProcessor.completeAnalysis(transformToModelAnalysisResult(response));
+        executorService.execute(() -> {
```
Comment (janinko) [inline rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/DeliverableAnalysisEndpointImpl.java:55]: please add `@RespondWithStatus(Response.Status.ACCEPTED)` to the interface method, so it returns 202 ACCEPTED instad of 204. And update the swagger doc acordingly.

File: rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/DeliverableAnalysisEndpointImpl.java [hunk 3]
```diff
@@ -38,9 +47,27 @@ public class DeliverableAnalysisEndpointImpl implements DeliverableAnalysisEndpo
...
+            ResultStatus result;
+            try {
+                resultProcessor.completeAnalysis(transformToModelAnalysisResult(response));
+                result = ResultStatus.SUCCESS;
+            } catch (RuntimeException e) {
+                log.error(
+                        String.format(
+                                "Storing results of deliverable operation with id=%s failed",
+                                response.getOperationId()),
+                        e);
```
Comment (janinko) [inline rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/DeliverableAnalysisEndpointImpl.java:65]: By default, SLF4J will use the latest parameter as a candidate for Throwable. If the provided parameter is an exception, SLF4J will print the complete stack trace in the log output.

For example, for a given logline:

log.info("App is running at {}, zone = {}, java version = {}, java vm = {}", LocalDateTime.now(), ZonedDateTime.now()
  .getZone(), System.getProperty("java.version"), System.getProperty("java.vm.name"), exceptionCause);
Copy
```suggestion
                log.error("Storing results of deliverable operation with id={} failed", response.getOperationId(), e);
```

**Review by janinko — CHANGES_REQUESTED** [review]

File: rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/DeliverableAnalysisEndpointImpl.java (hunk 2 above)
Comment (janinko) [reply rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/DeliverableAnalysisEndpointImpl.java:54]: It's supposed to go on the inerface

**Review by janinko — APPROVED** [review]

**Pull request merged**